network/server.go

```diff
@@ -588,12 +589,35 @@ func StringToAddrInfo(addr string) (*peer.AddrInfo, error) {
 	return addr1, nil
 }
 
+var (
+	// Regex used for matching loopback addresses (IPv4 and IPv6)
+	// This regex will match:
+	// /ip4/localhost/tcp/<port>
+	// /ip4/127.0.0.1/tcp/<port>
+	// /ip4/<any other loopback>/tcp/<port>
+	// /ip6/<any loopback>/tcp/<port>
+	loopbackRegex = regexp.MustCompile(
+		`/^\\/ip4\\/127(?:\\.[0-9]+){0,2}\\.[0-9]+\\/tcp\\/\\d+$|^\\/ip4\\/localhost\\/tcp\\/\\d+$|^\\/ip6\\/(?:0*\\:)*?:?0*1\\/tcp\\/\\d+$/gm`,
+	)
+)
+
 // AddrInfoToString converts an AddrInfo into a string representation that can be dialed from another node
 func AddrInfoToString(addr *peer.AddrInfo) string {
-	if len(addr.Addrs) != 1 {
-		panic("Not supported")
+	dialAddress := ""
+
+	// Find an address that's not a loopback address
+	for _, address := range addr.Addrs {
+		if !loopbackRegex.MatchString(address.String()) {
+			// Not a loopback address, dial address found
+			dialAddress = address.String()
```

Comment:
Just comments
(1) Is it ok if you don't exit loop after immediately you find suitable address?
(2) Addresses in addr.Addrs are always in same order? If not, AddrInfoToString returns different result sometimes for a peer.

Reply:
🤦‍♂️

Fixed
07ed343